mitmproxy/addons/script.py

```diff
@@ -30,9 +30,11 @@ def load_script(path: str) -> types.ModuleType:
         loader.exec_module(m)
         if not getattr(m, "name", None):
             m.name = path  # type: ignore
-        return m
+    except Exception as e:
+        ctx.log.error(str(e))
```

Comment:
A few comments here. First, the structure of this except bothers me - we will see an unbound local error for the m variable if there's an error somewhere before m is assigned. I think the right approach is to change the signature of the function to possibly return None, and then to make sure that m has a default value of None before entering the except handler. Second, we know the path here, so we should add that to the error message being logged, so the user knows which file had the error.



I think a better approach is to add another try/except just around the exec_module

Reply:
Thanks for the extensive review :) When I am printing the error as `e` , it also prints the filename and the lineno with it like `invalid syntax(example.py, line 4)`, then this is only for SyntaxError and I couldn't find a way to print the filename for other errors like `ValueError` mentioned in the original error. Even the traceback module didn't help :(